### ISSUE TITLE
Dataset names for NaturalEarthFeature() are not discoverable

### ISSUE THREAD

**Issue by ryneches** [SciTools/cartopy#2141]
### Description

The `NaturalEarthFeature()` interface takes a name for the dataset field. These names are munged from the filenames on the the [Natural Earth Data](https://www.naturalearthdata.com/) website. These names are not listed anywhere, nor is the way they are built into URLs described. The documentation simply gives an example without explanation :

'''
name:

    The name of the dataset, e.g. ‘admin_0_boundary_lines_land’.
'''

When an incorrect name is requested, `urllib` raises an `HTTPError`. The computed URL is not presented. This leaves the user with no way to diagnose the mistake they have made. The function definition contains the following doc string, which contains useful information that should be considered for inclusion in the user documentation :

    """
    Return the path to the requested natural earth shapefile,
    downloading and unzipping if necessary.

    To identify valid components for this function, either browse
    NaturalEarthData.com, or if you know what you are looking for, go to
    https://github.com/nvkelso/natural-earth-vector/tree/master/zips to
    see the actual files which will be downloaded.

    Note
    ----
        Some of the Natural Earth shapefiles have special features which are
        described in the name. For example, the 110m resolution
        "admin_0_countries" data also has a sibling shapefile called
        "admin_0_countries_lakes" which excludes lakes in the country
        outlines. For details of what is available refer to the Natural Earth
        website, and look at the "download" link target to identify
        appropriate names.

    """

This still does not explicitly describe how the URL is built, and so the user must still take a guess at the correct input. The `NEShpDownloader` class 

   FORMAT_KEYS = ('config', 'resolution', 'category', 'name')

    # Define the NaturalEarth URL template. Shapefiles are hosted on AWS since
    # 2021: https://github.com/nvkelso/natural-earth-vector/issues/445
    _NE_URL_TEMPLATE = ('https://naturalearth.s3.amazonaws.com/'
                        '{resolution}_{category}/ne_{resolution}_{name}.zip')

This scheme implicitly excludes some of the physical and cultural datasets :

```
110m_cultural/110m_cultural.zip
50m_physical/50m_physical.zip
110m_physical/110m_physical.zip
10m_physical/10m_physical.zip
10m_cultural/10m_cultural.zip
```

It also implicitly excludes all of the raster datasets :

```
10m_raster/GRAY_HR_SR.zip
10m_raster/GRAY_HR_SR_OB.zip
10m_raster/GRAY_HR_SR_OB_DR.zip
10m_raster/GRAY_HR_SR_W.zip
10m_raster/GRAY_LR_SR.zip
10m_raster/GRAY_LR_SR_OB.zip
10m_raster/GRAY_LR_SR_OB_DR.zip
10m_raster/GRAY_LR_SR_W.zip
10m_raster/HYP_HR.zip
10m_raster/HYP_HR_SR.zip
10m_raster/HYP_HR_SR_OB_DR.zip
10m_raster/HYP_HR_SR_W.zip
10m_raster/HYP_HR_SR_W_DR.zip
10m_raster/HYP_LR.zip
10m_raster/HYP_LR_SR.zip
10m_raster/HYP_LR_SR_OB_DR.zip
10m_raster/HYP_LR_SR_W.zip
10m_raster/HYP_LR_SR_W_DR.zip
10m_raster/NE1_HR_LC.zip
10m_raster/NE1_HR_LC_SR.zip
10m_raster/NE1_HR_LC_SR_W.zip
10m_raster/NE1_HR_LC_SR_W_DR.zip
10m_raster/NE1_LR_LC.zip
10m_raster/NE1_LR_LC_SR.zip
10m_raster/NE1_LR_LC_SR_W.zip
10m_raster/NE1_LR_LC_SR_W_DR.zip
10m_raster/NE2_HR_LC.zip
10m_raster/NE2_HR_LC_SR.zip
10m_raster/NE2_HR_LC_SR_W.zip
10m_raster/NE2_HR_LC_SR_W_DR.zip
10m_raster/NE2_LR_LC.zip
10m_raster/NE2_LR_LC_SR.zip
10m_raster/NE2_LR_LC_SR_W.zip
10m_raster/NE2_LR_LC_SR_W_DR.zip
10m_raster/OB_LR.zip
10m_raster/SR_HR.zip
10m_raster/SR_LR.zip
50m_raster/BATH_50M.zip
50m_raster/GRAY_50M_SR.zip
50m_raster/GRAY_50M_SR_OB.zip
50m_raster/GRAY_50M_SR_W.zip
50m_raster/HYP_50M_SR.zip
50m_raster/HYP_50M_SR_W.zip
50m_raster/MSR_50M.zip
50m_raster/NE1_50M_LC_SR.zip
50m_raster/NE1_50M_LC_SR_W.zip
50m_raster/NE1_50M_SR.zip
50m_raster/NE1_50M_SR_W.zip
50m_raster/NE2_50M_LC_SR.zip
50m_raster/NE2_50M_LC_SR_W.zip
50m_raster/NE2_50M_SR.zip
50m_raster/NE2_50M_SR_W.zip
50m_raster/OB_50M.zip
50m_raster/PRISMA_SR_50M.zip
50m_raster/SR_50M.zip
```

I'm not sure if this is intentional or not, but it is confusing as heck. I would certainly like to use the raster datasets, but because the 'ne_' prefix is hard-coded into the URL builder, there is no way get at them. In any event, here is a listing of all the datasets that can be accessed with the current interface :

[ne_names.txt](https://github.com/SciTools/cartopy/files/10856637/ne_names.txt)

Here is is a listing of all the Natural Earth datasets that are actually available :

[naturalearth.txt](https://github.com/SciTools/cartopy/files/10856578/naturalearth.txt)

I would recommend including this list somewhere in the user documentation. It would also be nice if the interface were updated to make it possible to access the raster datasets.

**Comment by dopplershift** [comment]
The documentation for the available map features could certainly be improved, though one challenge is that we don't control what's available, so some manual effort would be needed to keep it up to date. Pull requests improving the documentation and clarity would certainly be welcome.

Regarding the raster data sets, the `NaturalEarthFeature` (and really the base `Feature` class) is focused around vector features (with geometries and using a shapefile reader), so accessing the correctly-named feature is only one of many hurdles that stand in your way.

**Comment by greglucas** [comment]
We also have the undocumented `tools.cartopy_feature_download.py` script https://github.com/SciTools/cartopy/blob/main/tools/cartopy_feature_download.py
that iterates through a few of the combinations. It would definitely be welcome for someone to update our documentation around what is available!

Cartopy has RasterSource capabilities, so it would likely be figuring out how to extend some of those capabilities to the NaturalEarth Raster URLs. https://scitools.org.uk/cartopy/docs/latest/reference/generated/cartopy.io.RasterSource.html

**Comment by ryneches** [comment]
@dopplershift @greglucas If wrote a GitHub Action workflow to check the available datasets, what would you suggest for the interval and where would you want the output to live?

**Comment by greglucas** [comment]
I'm not sure what the GitHub Action is needed for? The NE datasets are updated pretty infrequently and really what we need is a listing in the docs for users to get a rough reference of. We don't want to be the defacto NaturalEarth source.

It looks like there are other places where people already gather some of the links: https://gist.github.com/DanielJWood/b71237cc200831acf8e637c05ce2c375
So you could use those, or do a few `aws s3 ls --no-sign-request s3://naturalearth/` on the object paths out there and copy/paste into the docs I think.

**Comment by philippemiron** [comment]
Similarly, I was trying to plot `Parks and protected lands` today, available [here](https://www.naturalearthdata.com/downloads/10m-cultural-vectors/parks-and-protected-lands/).

Here is a simple code.

```
import matplotlib.pyplot as plt
import cartopy.crs as ccrs

fig = plt.figure()
ax = fig.add_subplot(1, 1, 1, projection=ccrs.PlateCarree())

parks = cfeature.NaturalEarthFeature('cultural', 'parks_and_protected_lands', '10m')
ax.add_feature(parks)
```

It downloads the correct archive. But it fails because the shapefile in the archive does not have the same name.
```
.../cartopy/io/__init__.py:241: DownloadWarning: Downloading: https://naturalearth.s3.amazonaws.com/10m_cultural/ne_10m_parks_and_protected_lands.zip
  warnings.warn(f'Downloading: {url}', DownloadWarning)

KeyError: "There is no item named 'ne_10m_parks_and_protected_lands.shp' in the archive"
```
the shapefiles inside are actually:
- `ne_10m_parks_and_protected_lands_area.shp`
- `ne_10m_parks_and_protected_lands_line.shp`
- `ne_10m_parks_and_protected_lands_point.shp`
in that case.

Would it be possible to add an extra argument (or pass this in `**kwargs` for those datasets where the shapefile filenames do not match the archive .zip ?

**Comment by greglucas** [comment]
I think we would be open to ideas for how to improve the NaturalEarth interface. If there is some mapping to add to help here, or any of the suggestions above in this thread to list things that would probably be nice to add.

However, I will say that I am not sure what I would expect in this situation... You want a shapefile, but we get 3 shapefiles with areas, lines, and points, so would you want those all plotted together or as separate collections? This is where it might just be easier to document and point people at how to use the shapereader class to add the proper collection they want.

**Comment by philippemiron** [comment]
I agree that the expected behavior in that case is not clear. Couldn't instead of only throwing a `KeyError` exception, could also listing the available keys be a possibility?

**Comment by HathewayWill** [comment]
I would like to say that adding raster functionalitiy to the ccrs.feature.NaturalEarthFeature would be highly valueable to the scientific community.
